File: src/main/java/com/azbj/proposalde3/service/PreviousPolicyDetailsService.java
package com.azbj.proposalde3.service;

import com.azbj.proposalde3.repository.PreviousPolicyDetailsRepository;
import com.azbj.proposalde3.model.PreviousPolicyDetails;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class PreviousPolicyDetailsService {

    @Autowired
    private PreviousPolicyDetailsRepository previousPolicyDetailsRepository;

    public void deletePreviousPolicyRecord(String policyId) {
        previousPolicyDetailsRepository.deleteById(policyId);
    }

    public List<PreviousPolicyDetails> getPreviousPolicyDetails() {
        return previousPolicyDetailsRepository.findAll();
    }

    public PreviousPolicyDetails updatePreviousPolicyDetails(PreviousPolicyDetails previousPolicyDetails) {
        return previousPolicyDetailsRepository.save(previousPolicyDetails);
    }

    public byte[] getImage(String block, String item, String proposalType) {
        return previousPolicyDetailsRepository.findImage(block, item, proposalType);
    }

    public PreviousPolicyDetails fetchPreviousInsuranceCompanyInfo(String currentBlock, String currentItem) {
        return previousPolicyDetailsRepository.findByBlockAndItem(currentBlock, currentItem);
    }

    public PreviousPolicyDetails updatePreviousInsuranceCompanyInfo(String currentBlock, String currentItem, String newValue) {
        PreviousPolicyDetails details = previousPolicyDetailsRepository.findByBlockAndItem(currentBlock, currentItem);
        details.setInsuranceCompany(newValue);
        return previousPolicyDetailsRepository.save(details);
    }

    public String promptUser() {
        // Logic to prompt user with a question
        // This is a placeholder for actual implementation
        return "Do you wish to add more Policies?";
    }

    public PreviousPolicyDetails updateStatus(String status) {
        PreviousPolicyDetails details = new PreviousPolicyDetails();
        details.setStatus(status);
        return previousPolicyDetailsRepository.save(details);
    }
}

File: src/app/services/previous-policy-details.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class PreviousPolicyDetailsService {
  constructor(private http: HttpClient) {}

  deletePreviousPolicyRecord(policyId: string): Observable<any> {
    return this.http.delete(`/api/previous-policy-details/${policyId}`);
  }

  getImageField(currentItem: string): Observable<any> {
    return this.http.get(`/api/previous-policy-details/image?item=${currentItem}`);
  }

  fetchIPTypes(): Observable<string[]> {
    return this.http.get<string[]>('/api/previous-policy-details/ip-types');
  }

  getPreviousPolicyDetails(): Observable<any> {
    return this.http.get('/api/previous-policy-details');
  }

  updatePreviousPolicyDetails(previousPolicyDetails: any): Observable<any> {
    return this.http.post('/api/previous-policy-details', previousPolicyDetails);
  }

  getImage(block: string, item: string, proposalType: string): Observable<any> {
    return this.http.get(`/api/previous-policy-details/image?block=${block}&item=${item}&proposalType=${proposalType}`);
  }

  fetchPreviousInsuranceCompanyInfo(currentBlock: string, currentItem: string): Observable<any> {
    return this.http.get(`/api/previous-policy-details/insurance-company-info?block=${currentBlock}&item=${currentItem}`);
  }

  updatePreviousInsuranceCompanyInfo(currentBlock: string, currentItem: string, newValue: string): Observable<any> {
    return this.http.post('/api/previous-policy-details/insurance-company-info', { block: currentBlock, item: currentItem, value: newValue });
  }

  promptUser(): Observable<boolean> {
    return new Observable(observer => {
      const userResponse = confirm('Do you wish to add more Policies?');
      observer.next(userResponse);
      observer.complete();
    });
  }

  updateStatus(status: string): Observable<any> {
    return this.http.post('/api/previous-policy-details/status', { status });
  }
}
